openedx/features/philu_courseware/helpers.py

```diff
@@ -116,3 +124,38 @@ def is_post_assessment(section):
 
 def get_section_format(section):
     return section.format.lower() if section and section.format else ''
+
+
+def validate_problem_id(problem_id):
+    """
+    validate if problem_id is valid UsageKeyField or not
+    """
+    if not problem_id:
+        raise serializers.ValidationError({'problem_id': _('Problem id is required')})
+    try:
+        return UsageKey.from_string(problem_id)
+    except InvalidKeyError:
+        raise serializers.ValidationError({'problem_id': constants.INVALID_PROBLEM_ID_MSG})
+
+
+def revert_user_attempts_from_edx(course_id, user, problem_usage_key):
+    """
+    :param course_id: str course id of user
+    :param user: Django User
+    :param problem_usage_key: UsageKey problem id
+    """
+    course_id = CourseKey.from_string(course_id)
```

Comment:
What if `course_id` in the URL is invalid. Let's catch `InvalidKeyError` exception here as well